content/go/concepts/composition/composition.md

```diff
@@ -133,6 +133,129 @@ Margherita pizza is a medium pizza with toppings of: [tomatoes mozzarella basil]
 Pizzeria del Corso has a rating of 4 and serves the following pizzas: [{Margherita medium [tomatoes mozzarella basil] true}]
 ```
 
+## Inheritance through Composition
+
+A `struct` can 'inherit' the fields and methods of an embedded `struct` through composition rather than traditional [inheritance](https://www.codecademy.com/resources/docs/general/programming-paradigms/inheritance).
+
+```pseudo
+type Embedded struct {
+  field string
+}
+
+func (a Embedded) printField() string {
+  return a.field
+}
+
+type MainStruct struct {
+  Embedded
+}
+
+func PrintEmbedded(m MainStruct) string {
+  return m.printField()
+}
+```
+
+Consider the following addition to the Pizza Restaurant example:
+
+```go
+type Address struct {
+  Street  string
+  City    string
+  Country string
+}
+
+func (a Address) printAddress() string {
+  return a.Street + ", " + a.City
+}
+
+type Restaurant struct {
+  Name      string
+  Rating    int
+  PizzaMenu []Pizza
+  Address
+}
+
+func restaurantInfo(r Restaurant) string {
+  return r.Name + ", located at " + r.printAddress() + ", " + r.Country + " has a rating of " + fmt.Sprint(r.Rating) + " and serves the following pizzas: " + fmt.Sprint(r.PizzaMenu)
+}
+```
+
+An `Address` `struct` is created, and a `printAddress()` method takes an `Address` as its receiver. The anonymous field `Address` is added to `Restaurant` which allows access to the `Address` fields and methods.
+
+The `restaurantInfo()` function can now access the `printAddress()` method with `r.printAddress()` and the `Country` field with `r.Country`.
+
+The following example adds the `Address` `struct` and method:
+
+```go
```

Comment:
This code is getting is bloated. I think it's preferable to just show the main function and perhaps have a comment at the top that this builds on the previous code

Reply:
How about this wording for the comment?